trikotFoundation/src/iosMain/kotlin/com/mirego/trikot/foundation/concurrent/dispatchQueue/IOSSerialDispatchQueue.kt

```diff
@@ -20,12 +22,24 @@ open class IOSSerialDispatchQueue(identifier: String) : TrikotDispatchQueue {
         dispatch_async(
             serialQueue,
             freeze {
-                runQueueTask(block)
+                try {
+                    runQueueTask(block)
+                } catch (e: Throwable) {
+                    forwardExceptionOnMainThread(e)
+                }
             }
         )
     }
 
     private fun runQueueTask(block: DispatchBlock) {
         block()
     }
+
+    private fun forwardExceptionOnMainThread(exception: Throwable) {
+        if (!FoundationConfiguration.forwardsExceptionsOnMainThread) return
+
+        dispatch_async(dispatch_get_main_queue()) {
+            throw RuntimeException(exception)
```

Comment:
What about rethrowing without wrapping in RuntimeException?

Reply:
^This, I was thinking about adding an additional message to that RuntimeException, but I feel like throwing only the original exception is the right way to go. Thanks 👌🏻